src/Cake/Commands/DebugCommand.cs

```diff
@@ -38,15 +41,25 @@ public bool Execute(CakeOptions options)
                 throw new ArgumentNullException("options");
             }
 
-            var message = "Attach debugger to process {0} to continue";
+            var message = "Attach debugger to process {0} to continue or press <ESC> to skip and launch debugger upon first #break-directive.";
             var pid = _debugger.GetProcessId();
 
             _log.Debug("Performing debug...");
             _log.Information(Verbosity.Quiet, message, pid);
 
-            _debugger.WaitForAttach(Timeout.InfiniteTimeSpan);
-
-            _log.Debug("Debugger attached");
+            while (true)
+            {
+                if (_console.KeyAvailable && _console.ReadKey(true).Key == ConsoleKey.Escape)
```

Comment:
@patriksvensson the ```--debug``` flag is required for debugging (i.e. instructs composition to hook up RoslynDebugScriptSession which generates DLL + PDB and loads them into memory). In case we don't like to merge the different Roslyn script sessions, aborting is needed.